### PR TITLE
[0.3] Backport: Refactor wait package to be capped to 10m and simplify

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/cloud/aws/services/wait/wait.go [hunk 1]
```diff
@@ -32,61 +32,48 @@ import (
...
 	return wait.Backoff{
-		Duration: 2 * time.Second,
-		Factor:   1.5,
-		Jitter:   1.0,
-		Steps:    100,
+		Duration: time.Second,
+		Factor:   2,
+		Steps:    32,
+		Jitter:   4,
+		Cap:      20 * time.Second,
```
Comment (detiber) [inline pkg/cloud/aws/services/wait/wait.go]: It looks like Cap is not available in the version vendored for `release-0.3`

**Comment by vincepri** [comment]
@detiber Switched to PollImmediate with a default polling of 10s and maximum timeout of 10m

**Pull request merged**